pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -165,6 +170,18 @@ type PipelineTask struct {
 	Timeout *metav1.Duration `json:"timeout,omitempty"`
 }
 
+func (pt PipelineTask) validate() (errs *apis.FieldError) {
```

Comment:
I am *only* confused by this function. We have a `validate` and a `Validate` function now. I would either inline this in the exported `Validate` function *or* rename this function to something like `validateRefRoSpec`.

Reply:
I have renamed it for now to `validateRefOrSpec`. This function is the placeholder for all common validations e.g. validate workspaces. For now, its validating one of `ref` or `spec` is specified. As we refactor further, will keep on updating this function.